chcemvediet/apps/inforequests/admin.py

```diff
@@ -394,18 +410,62 @@ def delete_constraints(self, obj):
                 u'{} is the only action in the branch.'.format(admin_obj_format(obj))))
         return constraints
 
+    def can_snooze(self, obj):
+        if (obj.type in [Action.TYPES.EXPIRATION, Action.TYPES.APPEAL_EXPIRATION]
+                and obj.previous_action):
+            return True
+        return False
+
     def render_delete_form(self, request, context):
-        context[u'delete_constraints'] = self.delete_constraints(context[u'object'])
-        context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
+        obj = context[u'object']
+        context[u'delete_warnings'] = self.delete_warnings(obj)
+        context[u'delete_constraints'] = self.delete_constraints(obj)
+        if self.can_snooze(obj):
+            context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
         return super(ActionAdmin, self).render_delete_form(request, context)
 
+    @decorate(short_description=u'Delete selected actions')
+    @transaction.atomic
+    def delete_selected(self, request, queryset):
+        delete_warnings = []
+        delete_constraints = []
+        snoozed_actions = []
+        for obj in queryset:
+            delete_warnings += self.delete_warnings(obj, queryset)
+            delete_constraints += self.delete_constraints(obj)
+            if self.can_snooze(obj):
+                snoozed_actions.append(obj.previous_action)
```

Comment:
Pozor, tu v premennej `snoozed_actions` mas **predosle** akcie k mazanym akciam. Ale v template sa pise "Extend snooze of previous actions to the following deleted actions" a potom tam z `snoozed_actions` vypises tieto **predosle** akcie.

Bud zmen text v template, alebo v `snoozed_actions` (resp. inac nazvanej premennej) maj zoznam "deleted actions" ako pises v texte.

Reply:
@martinmacko47 Urcite som chcel vypisat na admin obrazovku tie akcie u ktorych sa predlzi snooze, cize previos_action. Bolo by zbytocne niektore akcie vypisovat 2*krat. Viac sa sa tam hodi slovicko "related", v zmysle akcii ktore oznacil admin na vymazanie do querysetu. :ok:

Reply:
Teraz je to lepsie naformulovane. Vdaka.